src/mappings/BoolMapping.cpp

```diff
@@ -46,10 +46,11 @@ BoolMapping::BoolMapping(const std::string& reference, RegisterMapping::Register
 bool BoolMapping::writeValue(bool value)
 {
     bool success;
+    const auto reader = getGroup()->getDevice()->getReader().lock();
```

Comment:
You need a null check when making a shared ptr from weak

Reply:
Added .expired() checks on all weak_ptrs.